### PR TITLE
Autodelete

### LINKED ISSUE #1: Delete command message
Add an option to auto-delete the message sent to run the command.

On decorators it's gonna look something like:

```js
@command(/^command$/i, {
    name: 'command',
    desc: 'My command',
    deleteMessage: true
})
```

It'll add something like

```js
function command (message) {
    if (deleteMessage && message.deletable) message.delete()
}
```

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
I just adding a new decorator "autodelete" to deleting automatically a command.

like this : 
```js
export default function autodelete() {
	return (target, key, descriptor) => {
		const wrapped = descriptor.value
		descriptor.value = function(...fnargs) {
			fnargs[0].delete().catch(console.error)
			return wrapped.apply(this, fnargs)
		}
		return descriptor
	}
}
```
You can use this decorator with 
```js
@autodelete()
```
you need to add this to a command.

Close #1

### CODE REVIEW

File: src/decorators/autodelete.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default function autodelete () {
+    return (target, key, descriptor) => {
+        const wrapped = descriptor.value
+		descriptor.value = function(...fnargs) {
+            fnargs[0].delete().catch(console.error)
```
Comment (DeltaEvo) [inline src/decorators/autodelete.js]: Remove `.catch(console.error)`

**Review by DeltaEvo — CHANGES_REQUESTED** [review]

**Comment by DeltaEvo** [comment]
Please rebase your changes on latest master, and squash your commits into only one :)

**Comment by DeltaEvo** [comment]
Close #1

**Review by cchudant — DISMISSED** [review]
You should use `Promise.all([fnargs[0].delete(), wrapped.apply(this, fnargs)` so that the promise rejection get caught if the bot lacks permission or if an unexpected error occurs (don't chain promises because we still want to call the command)

**Comment by lightdiscord** [comment]
Maybe check if a message is deletable before deleting it?
```js
message.deletable && message.delete()
```

**Review by DeltaEvo — APPROVED** [review]

**Pull request merged**